### PR TITLE
JAMES-2142 Add search in attachments in JMAP

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

**Comment by aduprat** [comment]
test this please

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 1]
```diff
@@ -1537,6 +1541,7 @@ public void getMessageListHasKeywordShouldIgnoreRecent() throws Exception {
     }
 
     @Test
+    @SuppressWarnings("unchecked")
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: there is a better way to fix these, see and cherry pick 8b37d572f2b4462f2c61f2a4b8d0413d05810a66

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/FilterToSearchQueryTest.java [hunk 2]
```diff
@@ -170,7 +170,8 @@ public void filterConditionShouldMapWhenText() {
                 SearchQuery.address(AddressType.Cc, text),
                 SearchQuery.address(AddressType.Bcc, text),
                 SearchQuery.headerContains("Subject", text),
-                SearchQuery.bodyContains(text))));
+                SearchQuery.bodyContains(text),
+                SearchQuery.attachmentContains(text))));
```
Comment (rouazana) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/FilterToSearchQueryTest.java]: that's strange to see a commit without test :) I guess integration test are following...

File: mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/InMemoryMailboxManager.java [hunk 3]
```diff
@@ -65,6 +66,12 @@ public InMemoryMailboxManager(MailboxSessionMapperFactory mailboxSessionMapperFa
     }
 
     @Override
+    @Inject
+    public void setMessageSearchIndex(MessageSearchIndex index) {
```
Comment (rouazana) [inline mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/InMemoryMailboxManager.java]: do you really need this? It is not used in this commit

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/FilterToSearchQueryTest.java (hunk 2 above)
Comment (aduprat) [reply server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/FilterToSearchQueryTest.java]: That's it

File: mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/InMemoryMailboxManager.java (hunk 3 above)
Comment (aduprat) [reply mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/InMemoryMailboxManager.java]: Yes, we have the same part of code in `CassandraMailboxManager`, so I just take it in order to be consistent.
This is a bad way to inject members, but modifying the constructor leads to a lot of changes...

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
Merge: https://github.com/linagora/james-project/pull/975

**Pull request closed without merging**